Generate.py

```diff
@@ -311,6 +316,19 @@ def handle_name(name: str, player: int, name_counter: Counter):
     return new_name
 
 
+def calculate_digit_length(slot_count: int):
+    # Should never run anything but the initial if statement, but on the off chance 'slot_count'
+    # is a non-positive number for some reason, the elif and else will at least prevent it from
+    # blowing up or throwing an exception.
+    if slot_count > 0:
+        digit_count = int(math.log10(slot_count)) + 1
+    elif slot_count == 0:
+        digit_count = 1
+    else:
+        digit_count = int(math.log10(-slot_count)) + 2
```

Comment:
Is there a reason why you are using log10 rather than converting to a string and taking the length? It seems like log10 is a bit much for this. 

Additionally, I believe that slot_count being non-positive would cause bigger issues somewhere else and should just throw an exception.

Reply:
`log10` can be more performant in some scenarios, but you're right that this is a relatively short number and is only performed once, so it should be simplified to that.

As for `slot_count`, in coming back to review the code again after receiving this comment, I noticed that anything less than 1 does nothing anyhow since `player` starts at 1 so the `while` loop started immediately after this new method is called would simply not execute if `args.multi` was anything less than 1. Given this context, a whole method may no longer be necessary as it could likely be replaced by a single line instead as any other values make those edge cases irrelevant.

I will try and pull up my old environment and test those changes before resubmitting.